### PR TITLE
Bug fix: Fixes unmarshal error for expected error in integration test…

### LINKED ISSUE #9819: Integration test framework - unmarshal error
**Describe the bug**
When using the integration `test` package under `istio.io/istio/mixer/pkg/adapter/test` and providing a `CallKind` of REPORT (and I assume CHECK), if the error provided is anything other than null, it results in an unmarshal failure.

```
	integration.go:254: Unable to unmarshal {
					"AdapterState": null,
					"Returns": [{
						"Check": {
							"Status": {},
							"ValidDuration": 0,
							"ValidUseCount": 0
						},
						"Quota": null,
						"Error": {
							"code": 2,
							"message": "1 error occurred:\n\t* rpc error: code = Unknown desc = report has not been successful\n\n"
						}
					}]
				} into Result: json: cannot unmarshal object into Go struct field Return.Error of type error
```

It appears the code is trying to unmarshal the "Error" block above into an error interface

**Expected behavior**
Expected the test to pass successfully or at least not fail on code internals

**Steps to reproduce the bug**
Provide an integration scenario with an error block. Example:

```json
{
   "AdapterState":null,
   "Returns":[
      {
         "Check":{
            "Status":{

            },
            "ValidDuration":0,
            "ValidUseCount":0
         },
         "Quota":null,
         "Error":{
            "code":2,
            "message":"1 error occurred:\n\t* rpc error: code = Unknown desc = report has not been successful\n\n"
         }
      }
   ]
}
```

**Version**
Istio - `1.0.0`

**Environment**
Mac OSX

### PR DESCRIPTION
**Pull request by PhilipGough**
… framework

Fixes #9819

### CODE REVIEW

File: mixer/pkg/adapter/test/integration.go [hunk 1]
```diff
@@ -118,7 +119,13 @@ type (
 		// Quota is the response from a check call to Mixer
 		Quota map[string]adapter.QuotaResult `json:"Quota"`
 		// Error is the error from call to Mixer
-		Error error `json:"Error"`
+		Error GrpcStatusErr `json:"Error"`
+	}
+
+	// GrpcStatusErr wraps the error returned from Mixer client
+	GrpcStatusErr struct {
```
Comment (douglas-reid) [inline mixer/pkg/adapter/test/integration.go]: any reason not to just use the Status struct directly?

Comment (PhilipGough) [reply mixer/pkg/adapter/test/integration.go]: The client returns a status which embeds another type which is actually what needs to be unmarshalled. It's not a problem to use that struct and I can make the change if preferred. I just thought this was less complex.

Comment (PhilipGough) [reply mixer/pkg/adapter/test/integration.go]: @douglas-reid I've made the request change. Not sure on the failing tests at this point.

**Pull request merged**